### ISSUE TITLE
GLOBAL IN does not use index

### ISSUE THREAD

**Issue by SaltTan** [ClickHouse/ClickHouse#5576]
**How to reproduce**
```
:) select count() from system.query_log where event_date in (today()) settings force_primary_key=1

1 rows in set. Elapsed: 0.004 sec.

:) select count() from system.query_log where event_date global in (today()) settings force_primary_key=1

Received exception from server (version 19.7.5):
Code: 277. DB::Exception: Received from localhost:9000, ::1. DB::Exception: Primary key (event_date, event_time) is not used and setting 'force_primary_key' is set..
```
**Expected behaviour**
GLOBAL IN to use index if possible.

**Comment by SaltTan** [comment]
A workaround for a distributed query is to put the values into a temporary table (for distributed query processing, temporary tables used in a query are passed to remote servers)
```
:) create temporary table tt as select today() as d;

:) select count() from system.query_log where event_date in tt settings force_primary_key=1

1 rows in set. Elapsed: 0.115 sec.
```

**Comment by alexey-milovidov** [comment]
Still relevant in 20.11.

**Comment by den-crane** [comment]
```
20.13.1.5273

select count() from system.query_log where event_date global in (today()) settings force_primary_key=1
┌─count()─┐
│     288 │
└─────────┘

1 rows in set. Elapsed: 0.002 sec.
```

fixed in #17253
